docs/apworld_dev_faq.md

```diff
@@ -43,3 +43,26 @@ A faster alternative to the `for` loop would be to use a [list comprehension](ht
 ```py
 item_pool += [self.create_filler() for _ in range(total_locations - len(item_pool))]
 ```
+
+---
+
+### I learned about indirect conditions in the world API document, but I want to know more. What are they and why are they necessary?
+
+The world API document mentions how to use `multiworld.register_indirect_condition` to register indirect conditions and **when** you should use them, but not *how* they work and *why* they are necessary. This is because the explanation is quite complicated.
+
+Region sweep (the algorithm that determines which regions are reachable) is a Breadth-First Search of the region graph from the origin region, checking entrances one by one and adding newly reached nodes (regions) and their entrances to the queue until there is nothing more to check.
+
+For performance reasons, AP only checks every entrance once. However, if any entrance's access_rule depends on region access, then it is possible for this to happen:
+1. An entrance that depends on a region is checked and determined to be nontraversable because the region hasn't been reached yet during the graph search.
+2. After that, the region is reached by the graph search.
+
+The entrance *would* now be determined to be traversable if it were rechecked, but it is not.
```

Comment:
```suggestion
This entrance *would* be called traversable if it were rechecked, but it won't be rechecked this cycle.
```

Reply:
"called traversable"? that doesn't seem like wording we use anywhere else maybe "marked traversable"? not sure
actually, maybe just "reachable"

Reply:
Yeahh, maybe just reachable would be fine. idk, reachable sounds weird to me, but it's probably okay